R/copernicus_article.R

```diff
@@ -77,18 +77,17 @@ copernicus_journals <- list(
   "Advances in Statistical Climatology, Meteorology and Oceanography" = "ascmo",
   "Annales Geophysicae" = "angeo",
   "Archives Animal Breeding" = "aab",
-  "ASTRA Proceedings" = "ap",
   "Atmospheric Chemistry and Physics" = "acp",
   "Atmospheric Measurement Techniques" = "amt",
   "Biogeosciences" = "bg",
   "Climate of the Past" = "cp",
   "DEUQUA Special Publications" = "deuquasp",
   "Drinking Water Engineering and Science" = "dwes",
-  "European Journal of Mineralogy" = "ejm",
   "Earth Surface Dynamics" = "esurf",
```

Comment:
Do we really need to deleted them ? 

This will probably break some code if one was using for example 
````r
rticles::copernicus_journal_abbreviations(journal_name = '"ASTRA Proceedings')
````

Can't we just keep them ? 

possibly in a separate list if we want to differentiate new and old journal name ?

Reply:
> If the journal is not known, the PDF still builds (I've have tested it with rticles + plain LaTeX). 

This is if the function `rticles::copernicus_journal_abbreviations` was not used to derive the journal abbreviation like in the template YAML header, right ? 

> If a journal is not published anymore by Copernicus, so it is unlikely that new articles are written for such journals.

Good point. 

But I like your warning solution to possibly warn there is something not expected. 

> I intended to keep the list precisely as Copernicus has it (easier to compare, and we do not drag old stuff with us)

What about keeping a list of old journal in a separate object and use this list to warn that this is an old journal ? 

What do you think ?

Reply:
> This is if the function rticles::copernicus_journal_abbreviations was not used to derive the journal abbreviation like in the template YAML header, right ?

No, I used the function, e.g., `r rticles::copernicus_journal_abbreviations(journal_name = 'gaga')` returns 
just something that is empty. More precise, the corresponding code returns `NULL`, which is then included in the template. 

> What about keeping a list of old journal in a separate object and use this list to warn that this is an old journal ?

There is just one problem. I would only know about the list now, and I have no idea what Copernicus did in the past, and I am not sure whether there is a record. Hence, I would prefer to keep that very general. What do you think?

Reply:
> More precise, the corresponding code returns NULL, which is then included in the template.

And what happens in that case when no journal is passed ? Just want to be sure to understand everything. THanks

Reply:
## Three scenarios: 

1. `rticles::copernicus_journal_abbreviations(journal_name = "")`: All journal abbreviations are passed onto LaTeX; looks odd but the PDF builds.
2. `rticles::copernicus_journal_abbreviations()`: Same as 1.
3. `rticles::copernicus_journal_abbreviations("non-existing journal")`: returns `NULL` and in the LaTeX nothing. 

### Scenario 1 and 2: Tex output

```tex
\documentclass[adgeo, ars, asr, ascmo, angeo, aab, acp, amt, bg, cp,
deuquasp, dwes, esurf, esd, essd, egqsj, ejm, fr, gchron, gh, gc, gi,
gmd, hgss, hess, jbji, jm, jsss, mr, ms, nhess, npg, os, pb, piahs,
sand, sd, soil, se, tc, we, wcd, we, wes, manuscript]{copernicus}
```

### Scenario 3: Tex output

```tex
\documentclass[, manuscript]{copernicus}
```

Bottom line, it is rather robust.

Reply:
Ok let's leave it that way then. Thanks for the explanation and the PR